graph/mongo/quadstore.go

```diff
@@ -113,38 +114,57 @@ func ensureIndexes(db *mgo.Database) error {
 }
 
 func createNewMongoGraph(addr string, options graph.Options) error {
-	conn, err := mgo.Dial(addr)
+	conn, err := dialMongo(addr, options)
 	if err != nil {
 		return err
 	}
 	defer conn.Close()
 	conn.SetSafe(&mgo.Safe{})
+	db := conn.DB("")
+	return ensureIndexes(db)
+}
+
+func dialMongo(addr string, options graph.Options) (*mgo.Session, error) {
+	var dialInfo mgo.DialInfo
+	dialInfo.Addrs = strings.Split(addr, ",")
+	user, ok, err := options.StringKey("username")
+	switch {
```

Comment:
Updated in the latest diff to if statements.  Also, @kortschak I think we need to return the empty struct rather than nil in this case.